lib/attr_json/record.rb

```diff
@@ -161,6 +161,33 @@ def attr_json(name, type, **options)
         # it with usual system will let simple_form and maybe others find it.
         if options[:rails_attribute]
           self.attribute name.to_sym, self.attr_json_registry.fetch(name).type
+
+          # Ensure the `attributes` are set to container attribute values after a
+          # record is found.
+          after_find do
+            config = self.class.attr_json_config
+            registry = self.class.attr_json_registry
+            cattr = config.default_container_attribute
+            next unless has_attribute?(cattr)
+            public_send(cattr).each do |key, value|
+              defn = registry.store_key_lookup(cattr, key)
+              attr_name = defn.try(:name)
+              self.send("#{attr_name}=", value) if attr_name
+            end
+            to_be_cleared = public_send(cattr).keys + [cattr.to_s]
+            to_be_cleared.select! { |k| has_attribute?(k) }
+            self.send(:clear_attribute_changes, to_be_cleared)
```

Comment:
I wonder if there's a simpler way to do this, could we use write_attribute here like you did below, instead of using the public accessor and then dealing with clearing changes?  

I think we need to add some lines to the tests, so the tests make sure nothing is marked `changed?` when it shouldn't be too?